### DISCUSSION TITLE
Issues with generating links to internally hosted pdf files.

### DISCUSSION THREAD

**Discussion by anoduck** [gethinode/hinode#1279] [Help] [answered]
Files are located `/hinode/static/logs/hpvTZlVNFnbWj_5GxF.pdf`, there are over 250 of them listed in a markdown table, which additionally provides meta information such as creation date, topic, etc.

`ERROR [en] REF_NOT_FOUND: Ref "logs/DRLl8B34a0hT3Bq2axthceFLG4E.pdf": "/home/hinode-template/content/docs/log-table.md:1:1": page not found`

__Variations__

Primarily, the following variations have been tried along with numerous combinations thereof:

| Attempt # | flavor         | Variety                                                                      |
|-----------|----------------|------------------------------------------------------------------------------|
| 1         | Standard Mkdwn | `[hpvTZlVNFnbWj_5GxF](/hinode/static/logs/hpvTZlVNFnbWj_5GxF.pdf)`             |
| 2         | Hugo Ref       | `[hpvTZlVNFnbWj_5GxF]({{% ref "/logs/hpvTZlVNFnbWj_5GxF.pdf" %}})`             |
| 3         | Hugo Ref       | `[hpvTZlVNFnbWj_5GxF]({{% ref "logs/hpvTZlVNFnbWj_5GxF.pdf" %}})`              |
| 4         | Hugo Rel       | `{{% ref "/logs/hpvTZlVNFnbWj_5GxF.pdf" %}}`                                   |
| 5         | Hugo Rel Ref   | `[hpvTZlVNFnbWj_5GxF]({{% relref "../../hpvTZlVNFnbWj_5GxF.pdf" %}})`          |
| 6         | Hugo Link      | `{{% link "/logs/hpvTZlVNFnbWj_5GxF.pdf" %}} hpvTZlVNFnbWj_5GxF {{% /link %}}` |

I have tried exchanging the "%" for "< >" to no resolve. 

This is the third time experiencing difficulties with this particular task, and there are two pages which use this type of formatting.

**Comment by markdumay** [comment, marked as answer]
Thanks for sharing the elaborate example. Let me clarify a few concepts first:

- [ref](https://gohugo.io/methods/shortcode/ref/) and [relref](https://gohugo.io/methods/shortcode/relref/) are shortcodes provided by **Hugo**. They resolve to the path of a **content page**.
- [link](https://gethinode.com/docs/components/link/) is a custom shortcode maintained by the **Hinode** theme. It resolves to both **content pages and site assets**.
- Hinode implements a render hook to handle **native markdown links** since release [v0.26.0](https://github.com/gethinode/hinode/releases/tag/v0.26.0).

Hugo treats content pages differently than (static) site assets and page resources. In general, **you cannot use `relref` and `ref`** to reference a static asset such as an `pdf` file. Hinode tries to simplify (unify) the handling of content pages and site assets. You can use standard Markdown links to reference both a content page and a site asset. 

> [!NOTE]
> Both `relref` and `ref` are not really necessary when using the Hinode theme. The documentation only mentions them for backwards compatibility.

Consider the following example. I have a static asset available on the path `./static/img/logo512x512.png`. I want the browser to open this file when I click on it. Hugo publishes `static` assets without the `static` base folder. You can use the following markdown link to reference this static file and let the browser decide what to do with it. Most browsers will simply display the image file.

```markdown
[Logo](/img/logo512x512.png)
```

I hope this clarifies the concept of links. Feel free to share your experiences and let us know how we can improve the documentation further.

**Reply by anoduck** [reply]
Big "AHA!" Thanks, Mark. It's crystal.
